sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py

```diff
@@ -91,68 +94,47 @@ def __init__(self, client: "EventHubProducerClient", target: str, **kwargs) -> N
         if partition:
             self._target += "/Partitions/" + partition
             self._name += "-partition{}".format(partition)
-        self._handler = None  # type: Optional[SendClientAsync]
-        self._outcome = None  # type: Optional[constants.MessageSendResult]
-        self._condition = None  # type: Optional[Exception]
+        self._handler: Optional[SendClientAsync] = None
+        self._outcome: Optional[constants.MessageSendResult] = None
+        self._condition: Optional[Exception] = None
         self._lock = asyncio.Lock(**self._internal_kwargs)
-        self._link_properties = {
-            types.AMQPSymbol(TIMEOUT_SYMBOL): types.AMQPLong(int(self._timeout * 1000))
-        }
+        self._link_properties = self._amqp_transport.create_link_properties(
+            {TIMEOUT_SYMBOL: int(self._timeout * 1000)}
+        )
+
 
     def _create_handler(self, auth: "JWTTokenAsync") -> None:
-        self._handler = SendClientAsync(
-            self._target,
+        self._handler = self._amqp_transport.create_send_client(
+            config=self._client._config,  # pylint:disable=protected-access
+            target=self._target,
             auth=auth,
-            debug=self._client._config.network_tracing,  # pylint:disable=protected-access
-            msg_timeout=self._timeout * 1000,
+            network_trace=self._client._config.network_tracing,  # pylint:disable=protected-access
             idle_timeout=self._idle_timeout,
-            error_policy=self._retry_policy,
+            retry_policy=self._retry_policy,
             keep_alive_interval=self._keep_alive,
             client_name=self._name,
             link_properties=self._link_properties,
             properties=create_properties(
-                self._client._config.user_agent  # pylint:disable=protected-access
+                self._client._config.user_agent,  # pylint: disable=protected-access
+                amqp_transport=self._amqp_transport,
             ),
-            **self._internal_kwargs
+            msg_timeout=self._timeout * 1000,
```

Comment:
ah, yep can update it! The pyamqp client doesn't take msg_timeout at all. Is that by design?

Reply:
Yes and no - it's because the client transport in pyamqp is full of holes and inaccuracies and undocumented refactors.
There's a good chance most of the kwargs we pass into the pyamqp client are either being ignored or are not working the way we think they are - so whenever you see things like this please mark withm with a big scary TODO so we remember to go back and see what's missing in pyamqp.
I wont have to time to investigate the pyamqp client in time for this pr - so we probably shouldn't block on it.